torch_xla/experimental/apply_layers.py

```diff
@@ -0,0 +1,72 @@
+from typing import Iterable
+
+import torch
+import torch.nn as nn
+from torch.utils._pytree import tree_map
+
+from torch_xla.experimental.scan import scan
+
+
+def apply_layers(layers: Iterable[torch.nn.Module], input_data):
```

Comment:
Note to self: The apply_layers function will check that the layers have identical structure in terms of weights and biases:
- They all have the same set of dictionary keys (parameter names)
- The parameters have the same shapes.